apps/studio/src/lib/db/clients/postgresql.ts

```diff
@@ -1257,11 +1259,33 @@ export class PostgresClient extends BasicDatabaseClient<QueryResult> {
     return result.rows[0]?.tableowner;
   }
 
-  protected async configDatabase(server: IDbConnectionServer, database: { database: string }) {
+  protected async configDatabase(server: IDbConnectionServer, database: { database: string}) {
+
+    let resolvedPw = null;
+    const redshiftOptions = server.config.redshiftOptions;
+
+    if (
+      server.config.client === "postgresql" &&
+      redshiftOptions?.iamAuthenticationEnabled
+    ) {
+      const nodeProviderChainCredentials = fromIni({
+        profile: redshiftOptions.awsProfile ?? "default",
+      });
+      const signer = new Signer({
```

Comment:
Question: Does this work with SSH tunnels?

Reply:
This does work via an SSH Tunnel, I tested locally ;)